friends/src/App.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from "react";
+import { Route, NavLink } from "react-router-dom";
+import axios from "axios";
+import "./App.css";
+import FriendsList from "./components/FriendList";
+import Form from "./components/Form";
+import Home from "./components/Home";
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: []
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get("http://localhost:5000/friends")
+      .then(response => {
+        console.log(response.data);
+        this.setState({ friends: response.data });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  addFriend = friend => {
+    axios
+      .post("http://localhost:5000/friends", friend)
+      .then(response => {
+        console.log(response.data);
+        this.setState({ friends: response.data });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  };
+
+  deleteFriend = id => {
+    axios
+      .delete(`http://localhost:5000/friends/${id}`)
+      .then(response => {
+        console.log(response.data);
+        this.setState({ friends: response.data });
+      })
+      .catch(err => {
+        console.log(err);
+      });
```

Comment:
The same applies for destructuring and the callback; this applies to your other .then and .catch callbacks too!